File: Assignment 01/Program 06/prog6.py
#Write a C program to find the minimum element in a 2D array in using dynamic memory allocation

import numpy as np

row = int(input("Enter the number of rows : "))
col = int(input("Enter the number of columns: "))

a = list(map(int,input("Enter the elements: ").split()))
mat1 = np.array(a)
mat2 = mat1.reshape(row,col)
print(mat2)

min = 100000000
for i in range(row):
    for j in range(col):
        if mat2[i][j] < min:
            min = mat2[i][j]

print("The minimum element :",min)

File: Data Structure and Algorithm/Assignment 01/Program 06/prog6.c
//Write a C program to find the minimum element in a 2D array in using dynamic memory allocation

#include<stdio.h>
#include<stdlib.h>
#include<limits.h>

int main(){
    int row,col;
    printf("Enter the no. of row and column respectively: ");
    scanf("%d %d",&row,&col);
    
    int **arr = (int**)malloc(row*sizeof(int*));

    for (int i = 0; i < row; i++)
    {
        arr[i] = (int*)malloc(col*sizeof(int));
    }
    
    for (int i = 0; i < row; i++)
    {
        for (int j = 0; j < col; j++)
        {
            printf("Enter the element [%d][%d]:",i+1,j+1);
            scanf("%d",&arr[i][j]);
        }
    }

    int min = INT_MAX;
    for (int i = 0; i < row; i++)
    {
        for (int j = 0; j < col; j++)
        {
            if(arr[i][j]< min)
                min = arr[i][j];
        }
        
    }
    printf("The maximum element is: %d",min);
    free(arr);
    return 0;
}